Sources/NIO/ByteBuffer-core.swift

```diff
@@ -383,11 +383,11 @@ public struct ByteBuffer {
         }
 
         var base = ensureCapacityAndReturnStorageBase(capacity: underestimatedByteCount)
-        var idx = 0
-        for b in bytes {
-            if idx >= underestimatedByteCount {
-                base = ensureCapacityAndReturnStorageBase(capacity: idx + 1)
-            }
+        var (iterator, idx) = UnsafeMutableBufferPointer(start: base, count: underestimatedByteCount).initialize(from: bytes)
+        assert(idx == underestimatedByteCount)
+        while let b = iterator.next() {
+            assert(S.self != String.UTF8View.self)
```

Comment:
@milseman I'm assuming `String.UTF8View` does never underestimate its count, right?

Reply:
Thank you @milseman. I made another PR (#392) to actually remove the assertion because I don't think we win anything having it...